src/NuGet.Core/NuGet.PackageManagement/PackageDownloader.cs

```diff
@@ -107,6 +126,18 @@ public static class PackageDownloader
 
                     foreach (var source in sourceGroup)
                     {
+                        if (nameSpaceLookupResult != null && nameSpaceLookupResult.PrefixMatch
```

Comment:
Why does prefix match mattrer? 
Would we do anything different if the match was the whole id?

Or is this supposed to mean that the look up has matched nothing?

Reply:
Maybe we can remove `Prefixmatch` I used as `nameSpaceLookupResult.PackageSources != null` logic, here I replaced this one with `nameSpaceLookupResult.PackageSources != null` logic.